android/app/src/main/java/org/smartregister/fhircore/fragment/PatientDetailFragment.kt

```diff
@@ -74,42 +86,91 @@ class PatientDetailFragment : Fragment() {
-      vaccineRecorded?.let { it1 ->
-        if (it1.isNotEmpty()) {
-          val tvVaccineRecorded = rootView.findViewById<TextView>(R.id.observation_detail)
-          tvVaccineRecorded?.text = "Received $it1 dose 1"
+    // load immunization data
+    viewModel.searchImmunizations(patitentId)
+
+    viewModel
+      .getImmunizations()
+      .observe(
+        viewLifecycleOwner,
+        Observer<List<Immunization>> {
+          if (it.isNotEmpty()) {
+            updateVaccineStatus(it)
+          }
+        }
+      )
+
+    return rootView
+  }
+
+  private fun updateVaccineStatus(immunizations: List<Immunization>) {
+
+    var isFullyVaccinated = false
+    viewModel.viewModelScope.launch {
+      immunizations.forEach { immunization ->
+        val doseNumber = (immunization.protocolApplied[0].doseNumber as PositiveIntType).value
+        if (isFullyVaccinated) {
+          return@forEach
+        }
+        val nextDoseNumber = doseNumber + 1
+        val vaccineDate = immunization.occurrenceDateTimeType.toHumanDisplay()
+        val nextVaccineDate = Utils.addDays(vaccineDate, 28)
+        val tvVaccineRecorded = rootView.findViewById<TextView>(R.id.vaccination_status)
+        tvVaccineRecorded.text =
+          resources.getString(
+            R.string.immunization_brief_text,
+            immunization.vaccineCode.text,
+            doseNumber
+          )
+
+        val tvVaccineSecondDose = rootView.findViewById<TextView>(R.id.vaccination_second_dose)
+        val btnRecordVaccine = activity?.findViewById<Button>(R.id.btn_record_vaccine)
+        tvVaccineSecondDose.visibility = View.VISIBLE
+        if (doseNumber == finalDoseNumber) {
+          isFullyVaccinated = true
+          tvVaccineRecorded.text = resources.getString(R.string.fully_vaccinated)
+          tvVaccineSecondDose.text = resources.getString(R.string.view_vaccine_certificate)
+          if (btnRecordVaccine != null) {
+            btnRecordVaccine.visibility = View.GONE
+          }
+        } else {
+          isFirstDoseAdministered = true
+          tvVaccineSecondDose.text =
+            resources.getString(
+              R.string.immunization_next_dose_text,
+              nextDoseNumber,
+              nextVaccineDate
+            )
         }
       }
     }
+  }
 
+  override fun onResume() {
+    super.onResume()
     // load immunization data
     viewModel.searchImmunizations(patitentId)
-
-    viewModel.liveSearchImmunization.observe(
-      viewLifecycleOwner,
-      {
-        adapter.submitList(it)
-        if (it.isNotEmpty()) {
-          val tvNoVaccinePlaceholder =
-            rootView.findViewById<TextView>(R.id.no_vaccination_placeholder)
-          tvNoVaccinePlaceholder.visibility = View.GONE
-          val noVaccinePlaceholder =
-            rootView.findViewById<View>(R.id.view_vaccination_status_separator)
-          noVaccinePlaceholder.visibility = View.GONE
+    viewModel
```

Comment:
This code block is not necessary, lines 92-111 will automatically update the UI once search is executed.

Reply:
updated